File: mvn-advent-of-code/src/test/java/advent/twenty_seventeen/Day22SporificaVirusTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day22SporificaVirusTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day22SporificaVirus("/twenty_seventeen/Day22-SporificaVirus-input.txt");

        assertEquals(5259, dp.getPart1Answer().intValue());
        assertEquals(2511722, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day22SporificaVirus("/twenty_seventeen/Day22-SporificaVirus-example#1.txt");

        assertEquals(5587, dp.getPart1Answer().intValue());
        assertEquals(2511944, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day22SporificaVirusTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day22SporificaVirusTest  extends FlatSpec with Matchers {
  "2017 Day 22 - Input" should "calculate answers" in {
    val dp = new Day22SporificaVirus("/twenty_seventeen/Day22-SporificaVirus-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day22SporificaVirus("/twenty_seventeen/Day22-SporificaVirus-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
